### ISSUE TITLE
InvalidTemplateParam on Eloquent Collection

### ISSUE THREAD

**Issue by mshamaseen** [psalm/psalm-plugin-laravel#232]
**Describe the bug**

InvalidTemplateParam error is returned when using Eloquent Collection methods (I've tried `find` and `unique`).

**Impacted Versions**

Latest; psalm/plugin-laravel                  v2.0.1

**Additional context**

```php
Route::get('test', function () {
    $collection = \Modules\Admin\Auth\Models\Admin::all();
    $test = $collection->find(1);
    echo $test->firstname ?? '';
});
```

ERROR: InvalidTemplateParam - routes/web.php:19:13 - Extended template param TModel of Illuminate\Database\Eloquent\Collection<Modules\Admin\Auth\Models\Admin|(TKey:Illuminate\Database\Eloquent\Collection as array-key)>&static expects type Illuminate\Database\Eloquent\Model, type Modules\Admin\Auth\Models\Admin|(TKey:Illuminate\Database\Eloquent\Collection as array-key) given (see https://psalm.dev/183)

**Comment by mr-feek** [comment]
I couldn't seem to reproduce this with the minimal example supplied. 

Could you please paste the entire output of `impacted versions` as requested in the template?

> Please paste the output of `composer show | grep -E 'psalm|laravel'` so we can see what versions are impacted?

I tested with the following deps:

```
barryvdh/laravel-ide-helper                    v2.9.1          Laravel IDE Helper, generates correct PHPDocs for all Facade classes, to improve auto-completion.
fruitcake/laravel-cors                         v2.0.4          Adds CORS (Cross-Origin Resource Sharing) headers support in your Laravel application
laravel/framework                              v8.75.0         The Laravel Framework.
laravel/serializable-closure                   v1.0.5          Laravel Serializable Closure provides an easy and secure way to serialize closures in PHP.
laravel/tinker                                 v2.6.3          Powerful REPL for the Laravel framework.
psalm/plugin-laravel                           v1.6.1          A Laravel plugin for Psalm
vimeo/psalm                                    v4.15.0         A static analysis tool for finding errors in PHP applications

```

**Comment by mshamaseen** [comment]
@mr-feek 

```
barryvdh/laravel-debugbar             v3.6.7    PHP Debugbar integration for Laravel
barryvdh/laravel-ide-helper           v2.12.3   Laravel IDE Helper, generates correct PHPDocs for all Facade classes, to improve auto-completion.
fruitcake/laravel-cors                v2.2.0    Adds CORS (Cross-Origin Resource Sharing) headers support in your Laravel application
laravel/framework                     v9.8.1    The Laravel Framework.
laravel/sail                          v1.13.9   Docker files for running a basic Laravel application.
laravel/sanctum                       v2.15.1   Laravel Sanctum provides a featherweight authentication system for SPAs and simple APIs.
laravel/serializable-closure          v1.1.1    Laravel Serializable Closure provides an easy and secure way to serialize closures in PHP.
laravel/telescope                     v4.8.3    An elegant debug assistant for the Laravel framework.
laravel/tinker                        v2.7.2    Powerful REPL for the Laravel framework.
laravel/ui                            v3.4.5    Laravel UI utilities and presets.
psalm/plugin-laravel                  v2.0.1    A Laravel plugin for Psalm
shamaseen/laravel-files-generator     0.5.1     Generate any kind of File/s from stubs with only a single command, literally, ANY TYPE.
shamaseen/laravel-repositories        1.1.2     repository pattern files generator
spatie/laravel-backup                 8.1.2     A Laravel package to backup your application
spatie/laravel-ignition               1.2.2     A beautiful error page for Laravel applications.
spatie/laravel-package-tools          1.11.3    Tools for creating Laravel packages
spatie/laravel-permission             5.5.2     Permission handling for Laravel 6.0 and up
spatie/laravel-ray                    1.29.7    Easily debug Laravel apps
spatie/laravel-signal-aware-command   1.2.0     Handle signals in artisan commands
vimeo/psalm                           4.22.0    A static analysis tool for finding errors in PHP applications
```

**Comment by lptn** [comment]
hey @mshamaseen

this should be fixed in v2.50, can you pls confirm?
